### ISSUE TITLE
Missing texts in HabboExperience animation

### ISSUE THREAD

**Issue by LauraWebdev** [ruffle-rs/ruffle#11999]
### Describe the bug

Hey there, I'm currently trying to port the "Habbo Experience" promotional animation from 2005. The original is still available at [http://images.habbo.com/downloads/experience/exp-20050704-1/index_uk.html](http://images.habbo.com/downloads/experience/exp-20050704-1/index_uk.html).

While porting, I've noticed that certain texts seem to be missing. There is no output in the console that would suggest a problem is present though.

Ruffle Version:
![image](https://github.com/ruffle-rs/ruffle/assets/7365286/8e38f72b-2ae9-4509-96e4-0cb9279952cf)
![image](https://github.com/ruffle-rs/ruffle/assets/7365286/568b8bd4-e231-4d34-b9c3-d17c26bf5756)
![image](https://github.com/ruffle-rs/ruffle/assets/7365286/bf8da497-2ce1-42c2-81d7-468e114c3c13)

### Expected behavior

The texts are displayed.

Flash Version:
![image](https://github.com/ruffle-rs/ruffle/assets/7365286/8708517c-8f76-449d-afc1-cd36f2b7a303)
![image](https://github.com/ruffle-rs/ruffle/assets/7365286/1f206142-82bc-4fff-82ea-24a622e090a6)
![image](https://github.com/ruffle-rs/ruffle/assets/7365286/03b78950-46e6-4ea4-a091-7522a258f0da)

### Affected platform

Self-hosted version

### Operating system

Windows 11 latest

### Browser

Microsoft Edge (Chromium based) latest

### Additional information

The swf loads certain additional files on startup.
- scene3.swf
- scene4.swf
- scene5.swf
- scene6.swf
- data
  - conf.xml
  - [language-code]
    - console.swf
    - localization.xml
    - structure.xml

I've confirmed that the xml files are valid and the project is still working using native flash in Pale Moon. The missing texts would be in localization.xml and console.swf, however the texts in structure.xml are used properly.

This problem exists both in the self-hosted version and the browser plugin.

**Comment by Dinnerbone** [comment]
Looks like we need to migrate XMLNode to NativeObject. Part of #701

**Comment by LauraWebdev** [comment]
Thanks for checking it out! It's interesting this only happens to one half of the UI. Does it use a different way to display the texts within structure.xml compared to localization.xml/console.swf?

**Comment by Dinnerbone** [comment]
Yeah, for the "Scene" stuff (ie character texts) it seems to just use `firstChild.childNodes`, whereas for localisation it uses a custom `findNode` to try and get it. We support the first no problem, but when it's trying to change the definition of what XMLNode is to add its helper method there, we fail :(
